### PR TITLE
RNA_DNA_classes.py added in feature

### PR DESCRIPTION
**Pull request by kosolapova**

### CODE REVIEW

**Review by DariaKIL — COMMENTED** [review]
It is necessary to consider lowercase symbols in `gc()` function + add multiplication by 100%

**Review by DariaKIL — COMMENTED** [review]
`for base in sequence` 
Will be better to add `self` before `sequence`

File: RNA_DNA_classes.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    base_pairs = {'A': 'U', 'U': 'A',  'G': 'C', 'C': 'G', 'a': 'u', 'u': 'a',  'g': 'c', 'c': 'g'}
+
+    def __init__(self, sequence):
+        self.sequence = sequence
+        for base in self.sequence:
+            if base not in self.base_pairs:
+                raise ValueError('RNA sequence can contain only A, U, G and C bases')
+
+    def gc(self):
+        return (self.count('G') + self.count('C'))/len(self)
```
Comment (DariaKIL) [inline RNA_DNA_classes.py]: It is necessary to consider lowercase symbols in `gc()` function + add multiplication by 100%

File: RNA_DNA_classes.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            complement += self.base_pairs[base]
+        return complement[::-1]
+
+
+class Dna(Rna):
+    base_pairs = {'A': 'T', 'T': 'A', 'G': 'C', 'C': 'G', 'a': 't', 't': 'a', 'g': 'c', 'c': 'g'}
+
+    def __init__(self, sequence):
+        self.sequence = sequence
+        for base in sequence:
```
Comment (DariaKIL) [inline RNA_DNA_classes.py]: Will be better to add self before sequence

File: RNA_DNA_classes.py [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
 
-    def __init__(self, sequence):
+    def __init__(self):
         self.sequence = sequence
         for base in self.sequence:
             if base not in self.base_pairs:
                 raise ValueError('RNA sequence can contain only A, U, G and C bases')
 
     def gc(self):
-        return (self.count('G') + self.count('C'))/len(self)
```
Comment (DariaKIL) [inline RNA_DNA_classes.py]: Ok, now it's work right 👍

**Review by DariaKIL — APPROVED** [review]

**Comment by agbragin** [comment]
You should not close and merge before the teacher has a chance to review the code!

File: RNA_DNA_classes.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        self.sequence = sequence
+        for base in self.sequence:
+            if base not in self.base_pairs:
+                raise ValueError('RNA sequence can contain only A, U, G and C bases')
+
+    def gc(self):
+        return (self.count('G')+self.count('g')+self.count('C')+self.count('c')) / len(self) * 100
+
+    def reverse_complement(self):
+        complement = ''
```
Comment (agbragin) [inline RNA_DNA_classes.py:14]: Try: `Dna('GATC').reverse_complement().reverse_complement()`

File: RNA_DNA_classes.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        for base in self:
+            complement += self.base_pairs[base]
+        return complement[::-1]
+
+
+class Dna(Rna):
```
Comment (agbragin) [inline RNA_DNA_classes.py:20]: Try: `Dna('GATC') == Dna('gatc')`

File: RNA_DNA_classes.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    base_pairs = {'A': 'T', 'T': 'A', 'G': 'C', 'C': 'G', 'a': 't', 't': 'a', 'g': 'c', 'c': 'g'}
+
+    def __init__(self, sequence):
+        self.sequence = sequence
+        for base in self.sequence:
+            if base not in self.base_pairs:
+                raise ValueError('DNA sequence can contain only A, T, G and C bases')
+
+    def transcribe(self):
```
Comment (agbragin) [inline RNA_DNA_classes.py:29]: Do we treat DNA as sense or antisense? If the second one, why we do not reverse the sequence?

**Review by agbragin — COMMENTED** [review]
See the comments.

File: RNA_DNA_classes.py (hunk 4 above)
Comment (kosolapova) [reply RNA_DNA_classes.py:14]: As I've already closed pull request and merged, I couldn't make new commit here, could I?

See your point. One of possibilities to make fix it is to `return Rna(complement[::-1])` in Rna class, for Dna class - duplicate that function with `return Dna(complement[::-1])`

File: RNA_DNA_classes.py (hunk 6 above)
Comment (kosolapova) [reply RNA_DNA_classes.py:29]: Usually we expect that DNA would be treated as sense strand, so changing 'T' to 'U' would be enough.
But if it is antisense (as I treated it), yes, reversing is necessary as nucleic acids should be written in 5'-3' way.

**Pull request merged**